tests/components/sensor/test_device_automation.py

```diff
@@ -0,0 +1,580 @@
+"""The test for sensor device automation."""
+import pytest
+
+from homeassistant.components.sensor import DOMAIN, DEVICE_CLASSES
+from homeassistant.components.sensor.device_automation import (
+    ENTITY_CONDITIONS,
+    ENTITY_TRIGGERS,
+)
+from homeassistant.const import STATE_UNKNOWN, CONF_PLATFORM
+from homeassistant.setup import async_setup_component
+import homeassistant.components.automation as automation
+from homeassistant.components.device_automation import (
+    _async_get_device_automations as async_get_device_automations,
+)
+from homeassistant.helpers import device_registry
+
+from tests.common import (
+    MockConfigEntry,
+    async_mock_service,
+    mock_device_registry,
+    mock_registry,
+)
+
+
+@pytest.fixture
+def device_reg(hass):
+    """Return an empty, loaded, registry."""
+    return mock_device_registry(hass)
+
+
+@pytest.fixture
+def entity_reg(hass):
+    """Return an empty, loaded, registry."""
+    return mock_registry(hass)
+
+
+@pytest.fixture
+def calls(hass):
+    """Track calls to a mock serivce."""
+    return async_mock_service(hass, "test", "automation")
+
+
+def _same_lists(a, b):
+    if len(a) != len(b):
```

Comment:
Maybe use the `Counter` class as in this answer:
https://stackoverflow.com/a/9623147

Reply:
I'm using it to compare lists of dicts, so naively replacing:
`assert _same_lists(conditions, expected_conditions)`
with:
`assert collections.Counter(conditions) == collections.Counter(expected_conditions)`

Throws `TypeError: unhashable type: 'dict'`

Did I misunderstand what you meant?

Reply:
No sorry. I didn't read beyond this function to where it's used.